Monika After Story/game/zz_calendar.rpy

```diff
@@ -0,0 +1,1695 @@
+        """
+        Removes an event from the calendar.
+
+        SEE removeEvent_el for important NOTES regarding this function.
+
+        IN:
+            ev - event to remove
+            month - SEE removeEvent_el
+            day - SEE removeEvent_el
+            remove_all - SEE removeEvent_el
+        """
+        removeEvent_el(
+            ev.eventlabel,
+            month=month,
+            day=day,
+            remove_all=remove_all
+        )
+
+
+    def removeRepeatable(identifier, month=None, day=None):
+        """
+        Removes a repeatable from the calendar.
+
+        NOTE: The default params will check EVERY SINGLE calendar spot for the
+        repeatable to remove. It is considered HIGHLY INEFFICIENT. Try to use
+        the other removeRepeatable functions if possible, or narrow the search
+        using month and day.
+
+        IN:
+            identifier - identifier of the repeatable to remove
+            month - If given (and a valid month), will only check the calendar
+                in the given month.
+                (Default: None)
+            day - If given (and a valid day), will only check the calendar for
+                the given day for reach month
+                (Default: None)
+        """
+        # inital sanity checks
+        if month not in range(1,13):
+            month = None
+
+        if day not in range(1,32):
+            day = None
+
+        # now to see which operation we are doing
+        if month is not None and day is not None:
+            # ideally we want the user to pass in a month and day
+            _removeRepeatable_md(identifier, month, day)
+
+        elif month is not None:
+            # probably common to clean a month
+            _removeRepeatable_m(identifier, month)
+
+        elif day is not None:
+            # less common to clean a particular day of a month
+            _removeRepeatable_d(identifier, day)
+
+        else:
+            # full scan, hopefully no one does this
+            _removeRepeatable(identifier)
+
+
+    def removeRepeatable_d(identifier, _date):
+        """
+        Removes a repeatable from teh calendar at a precise datetime.date
+
+        IN:
+            identifier - identifier of the repeatable to remove
+            _date - datetime.date we should look at for removal
+        """
+        _removeRepeatable_md(identifier, _date.month, _date.day)
+
+
+    def removeRepeatable_dt(identifier, _datetime):
+        """
+        Removes a repeatable from teh calendar at aprecise datetime
+
+        IN:
+            identifier - identifier of the repeatable to remove
+            _datetime - datetime we should look at for removal
+        """
+        removeRepeatable_d(identifier, _datetime.date())
+
+
+# add repeatable events
+init python:
+
+    import store.mas_calendar as calendar
+
+    calendar.addRepeatable("New years day","New years day",month=1,day=1,year_param=list())
```

Comment:
All of the words in the Holidays should be capitalized, so "New Year's Day, Valentine's Day, White Day, Christmas Eve, Christmas Day, New Year's Eve.